source/common/http/conn_manager_impl.cc

```diff
@@ -491,28 +491,43 @@ const Network::Connection* ConnectionManagerImpl::ActiveStream::connection() {
   return &connection_manager_.read_callbacks_->connection();
 }
 
+// Ordering in this function is complicated, but important.
+//
+// We want to do minimal work before selecting route and creating a filter
+// chain to maximize the number of requests which get custom filter behavior,
+// e.g. registering access logging.
```

Comment:
I'm a little confused so I think I'm missing something. What type of access logging are you referring to exactly? Encoding  filters are not required for doing access logging. Base access logging happens on stream destruction and AFAIK local replies should have the response code logged in the base stream info?